eq/client/compositor.cpp

```diff
@@ -1293,14 +1302,168 @@ void Compositor::_drawPixels( const Image* image, const ImageOp& op,
-        glDepthMask( true );
+        EQ_GL_CALL( glDepthMask( true ))
     else
     {
         const ColorMask& colorMask = op.channel->getDrawBufferMask();
-        glColorMask( colorMask.red, colorMask.green, colorMask.blue, true );
+        EQ_GL_CALL( glColorMask( colorMask.red, colorMask.green, colorMask.blue,
+                                 true ));
+    }
+}
+
+void Compositor::_drawPixels_GLSL( const Image* image, const ImageOp& op,
+                                   const Frame::Buffer which )
+{
+    const PixelViewport& pvp = image->getPixelViewport();
+    LBLOG( LOG_ASSEMBLY ) << "_drawPixels_GLSL " << pvp << " offset "
+                          << op.offset << std::endl;
+
+    Channel* channel = op.channel; // needed for glewGetContext
+    const util::Texture* texture = 0;
+    if( image->getStorageType() == Frame::TYPE_MEMORY )
+    {
+        LBASSERT( image->hasPixelData( which ));        
+        util::ObjectManager& objects = channel->getObjectManager();
+
+        util::Texture* ncTexture = objects.obtainEqTexture(
+            which == Frame::BUFFER_COLOR ? colorDBKey : depthDBKey,
+            GL_TEXTURE_RECTANGLE_ARB );
+        texture = ncTexture;
+
+        const Vector2i offset( -pvp.x, -pvp.y ); // will be applied with quad
+        image->upload( which, ncTexture, offset, objects );
+    }
+    else // texture image
+    {
+        LBASSERT( image->hasTextureData( which ));
+        texture = &image->getTexture( which );
+    }
+
+    const GLint textureUnit = 0;
+    EQ_GL_CALL( glActiveTexture( GL_TEXTURE0 + textureUnit ));
+    texture->bind();
+    texture->applyZoomFilter( op.zoomFilter );
+    texture->applyWrap();
+
+    if ( which == Frame::BUFFER_COLOR )
+        EQ_GL_CALL( glDepthMask( false ))
+    else
+    {
+        LBASSERT( which == Frame::BUFFER_DEPTH )
+        EQ_GL_CALL( glColorMask( false, false, false, false ));
+    }
+
+    const float startX = float
+        ( op.offset.x() + pvp.x * op.pixel.w + op.pixel.x );
+    const float endX   = float
+        ( op.offset.x() + pvp.x + pvp.w * op.pixel.w*op.zoom.x() + op.pixel.x );
+    const float startY = float
+        ( op.offset.y() + pvp.y * op.pixel.h + op.pixel.y );
+    const float endY   = float
+        ( op.offset.y() + pvp.y + pvp.h * op.pixel.h*op.zoom.y() + op.pixel.y );
+
+    util::ObjectManager& om = channel->getObjectManager();
+    GLuint program = om.getProgram( shader2DKey );
+    GLuint vertexArray = om.getVertexArray( shader2DKey );
+    GLuint vertexBuffer = om.getBuffer( shader2DKey );
+    GLuint uvBuffer = om.getBuffer( shader2DKey + 1 );
+    if( program == util::ObjectManager::INVALID )
+    {
+        vertexBuffer = om.newBuffer( shader2DKey );
+        uvBuffer = om.newBuffer( shader2DKey + 1 );
+        vertexArray = om.newVertexArray( shader2DKey );
+        program = om.newProgram( shader2DKey );
+
+        const char* vertexShader_glsl = {
+            "#version 330 core\n"
```

Comment:
No. GL 2.0 is 110 GLSL. Need to check if the shader works with that. Higher GL version otherwise.